model/project_ref.go

```diff
@@ -1029,9 +1029,25 @@ func mergeBranchAndRepoSettings(pRef *ProjectRef, repoRef *RepoRef) (*ProjectRef
 	reflectedRepo := reflect.ValueOf(repoRef).Elem().Field(0) // specifically references the ProjectRef part of RepoRef
 
 	util.RecursivelySetUndefinedFields(reflectedBranch, reflectedRepo)
+
+	// Include Parsley filters defined at repo level alongside project filters.
+	mergeParsleyFilters(pRef, repoRef)
+
 	return pRef, err
 }
 
+func mergeParsleyFilters(pRef *ProjectRef, repoRef *RepoRef) {
+	if repoRef.ParsleyFilters == nil || len(repoRef.ParsleyFilters) == 0 {
+		return
+	}
+
+	if pRef.ParsleyFilters == nil {
+		pRef.ParsleyFilters = []ParsleyFilter{}
+	}
+
+	pRef.ParsleyFilters = append(pRef.ParsleyFilters, repoRef.ParsleyFilters...)
```

Comment:
Can we write a test where just repoRef filters are set? I'm wondering if we're going to end up accidentally duplicating them because we'll have set the pRef's undefined fields at line 1031

Reply:
Good catch, switching the order of `mergeParsleyVariables` and `RecursivelySetUndefinedFields` fixes this.